src/dda/env/dev/types/linux_container.py

```diff
@@ -81,7 +89,10 @@ def start(self) -> None:
             from dda.utils.retry import wait_for
 
             if not self.config.no_pull:
-                self.docker.wait(["pull", self.config.image], message=f"Pulling image: {self.config.image}")
+                pull_command = ["pull", self.config.image]
+                if self.config.arch is not None:
+                    pull_command.extend(["--platform", f"linux/{self.config.arch}"])
```

Comment:
Inconsequential performance nit:

```suggestion
                    pull_command.extend(("--platform", f"linux/{self.config.arch}"))
```